fuel-vm/src/interpreter/blockchain.rs

```diff
@@ -108,9 +108,11 @@ where
 
         // update frame pointer, if we have a stack frame (e.g. fp > 0)
         if fp > 0 {
-            let fpx = add_usize(fp, CallFrame::code_size_offset());
+            let fp_code_size = add_usize(fp, CallFrame::code_size_offset());
+            let fp_code_size_end = add_usize(fp_code_size, WORD_SIZE);
 
-            self.memory[fp..fpx].copy_from_slice(&length.to_be_bytes());
+            let length = length as Word;
```

Comment:
> https://github.com/FuelLabs/fuel-specs/blob/master/src/vm/instruction_set.md#ldc-load-code-from-an-external-contract
Shouldn't the length of LDC be added to the existing size of the contract instead of replacing it?

pseudocode suggestion:
```suggestion
            let length = Word::from_be_bytes(self.memory[fp_code_size..fp_code_size_end]).checked_add(length as word).ok_or_else(|| PanicReason::MemoryOverflow.into())?;
```